limited_area/mesh.py

```diff
@@ -275,17 +303,18 @@ def subset_fields(self,
 
         # Make boundary Mask's between 0 and the number of specified relaxation
         # layers
-        region.mesh.createVariable('bdyMaskCell', 'i4', ('nCells',))
-        region.mesh.createVariable('bdyMaskEdge', 'i4', ('nEdges',))
-        region.mesh.createVariable('bdyMaskVertex', 'i4', ('nVertices')) 
+        if self.check_grid():
```

Comment:
Is this check necessary? Hasn't the MeshHandler instance in self already been checked by check_grid() before subset_grid is called?